### PR TITLE
Refactor Network Manager header file

### PR DESCRIPTION
**Pull request by ravibhagavandas**
Description
-----------
Split network manager to decouple network types ( used by low level drivers ) from network manager data structures

Checklist:
----------

-[ X ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

File: demos/common/include/iot_ble_numericComparison.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #ifndef _AWS_BLE_NUMERIC_COMPARISON_H_
 #define _AWS_BLE_NUMERIC_COMPARISON_H_
 
+#include "queue.h"
```
Comment (lt72) [inline demos/common/include/iot_ble_numericComparison.h]: What s this for? "queue.h" should not be needed

File: demos/common/network_manager/aws_iot_network_manager.c [hunk 2]
```diff
@@ -57,12 +61,12 @@
 /**
  * @brief Initializer for WiFi Network Info structure.
  */
-#define _NETWORK_INFO_WIFI            { .ulNetworkType = AWSIOT_NETWORK_TYPE_WIFI, .xNetworkState = eNetworkStateUnknown }
+#define _NETWORK_INFO_WIFI            { .ulNetworkType = AWSIOT_NETWORK_TYPE_WIFI, .xNetworkList = IOT_LINK_INITIALIZER, .xNetworkState = eNetworkStateUnknown }
```
Comment (lt72) [inline demos/common/network_manager/aws_iot_network_manager.c]: per coding conventions, no Hungarian notation. So please remove the 'x', 'ul' and so on.... look at task pol or any mqtt file to get an idea...

File: demos/common/network_manager/aws_iot_network_manager.c [hunk 3]
```diff
@@ -106,8 +110,8 @@ typedef struct NMNetworkInfo
 typedef struct NetworkManagerInfo
 {
     SemaphoreHandle_t xSubscriptionLock;
-    Link_t xSubscriptionListHead;
-    Link_t xNetworkListHead;
+    IotListDouble_t xSubscriptionListHead;
+    IotListDouble_t xNetworkListHead;
     IotTaskPool_t* pxTaskPool;
```
Comment (lt72) [inline demos/common/network_manager/aws_iot_network_manager.c]: I guess you are using the task pool pointer to initialize network maanger to use one specific task pool, correct?  if so, that is OK and you should use the system task pool via the handle #define'd in the task pool types header...
the max concurrent jobs is useless IMO, we should remove it. Just create dynamic tasks and recycle them in the callback.

File: demos/common/network_manager/aws_iot_network_manager.c (hunk 2 above)
Comment (ravibhagavandas) [reply demos/common/network_manager/aws_iot_network_manager.c]: Updated in PR https://github.com/aws/amazon-freertos/pull/462

File: demos/common/network_manager/aws_iot_network_manager.c (hunk 3 above)
Comment (ravibhagavandas) [reply demos/common/network_manager/aws_iot_network_manager.c]: Updated in PR https://github.com/aws/amazon-freertos/pull/462

File: demos/common/include/iot_ble_numericComparison.h (hunk 1 above)
Comment (ravibhagavandas) [reply demos/common/include/iot_ble_numericComparison.h]: Its a dependency in the for numeric comparison queue.

**Pull request merged**